### PR TITLE
add unit test for generator

### PR DESCRIPTION
**Pull request by zyjiaobj**
# Changes

- fix channel block bug when generate/clean resources are empty
- add unit tests for generator
- update `go-bindata` build path and generated htmltemplatebindata.go file
- update `go_test` to test all go files

### CODE REVIEW

File: pkg/generator/batch_generator_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+func TestBatchGenerator(t *testing.T) {
+
+	generateFunc := func(ns string, index int) (string, string) {
+		return ns, fmt.Sprintf("%s-%d", ns, index)
+	}
+	postGeneratorFunc := func(ns, name string) error {
+		return nil
+	}
+
+	t.Run("no need to create ksvc", func(t *testing.T) {
```
Comment (qibobo) [inline pkg/generator/batch_generator_test.go:35]: Please provide more detail message, just "no need to create ksvc" is not very clear.

File: pkg/generator/batch_generator_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		// should complete immediately
+		assert.Assert(t, duration < 1)
+	})
+
+	t.Run("should complete in 4s", func(t *testing.T) {
+		start := time.Now().Unix()
+		generator.NewBatchGenerator(time.Duration(1)*time.Second, 8, 2, 2, []string{"ns1", "ns2"}, generateFunc, postGeneratorFunc).Generate()
+		duration := time.Now().Unix() - start
+		// should complete in count/batch = 4 seconds
+		assert.Assert(t, duration >= 4 && duration <= 5)
```
Comment (qibobo) [inline pkg/generator/batch_generator_test.go:48]: It may need to assert that the generateFunc is called 8 times.

Comment (zyjiaobj) [reply pkg/generator/batch_generator_test.go:48]: good point

File: pkg/generator/batch_cleaner_test.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	"testing"
+	"time"
+
+	"gotest.tools/assert"
+	"knative.dev/kperf/pkg/generator"
+)
+
+func TestBatchCleaner(t *testing.T) {
+
+	var cleanFuncCaled uint64
```
Comment (zhanggbj) [inline pkg/generator/batch_cleaner_test.go:28]: perhaps a typo cleanFuncCaled-->  cleanFuncCalled

**Comment by zhanggbj** [comment]
@zyjiaobj Look neat! just left some minor comments. Thanks!

**Comment by zyjiaobj** [comment]
@zhanggbj All resolved, thanks.

**Comment by zhanggbj** [comment]
/lgtm

@zyjiaobj Cool! Merging it.

**Review by zhanggbj — APPROVED** [review]

**Pull request merged**